model_api/cpp/models/src/detection_model.cpp

```diff
@@ -39,135 +39,52 @@ DetectionModel::DetectionModel(const std::string& modelFile,
 
-std::unique_ptr<DetectionModel> DetectionModel::create_model(const std::string& modelFile, std::shared_ptr<InferenceAdapter> adapter, std::string model_type, const ov::AnyMap& configuration) {
-    std::shared_ptr<ov::Model> model = ov::Core{}.read_model(modelFile);
-    if (model_type.empty()) {
-        model_type = model->get_rt_info<std::string>("model_info", "model_type");
-    }
+DetectionModel::DetectionModel(std::shared_ptr<ov::Model>& model, const ov::AnyMap& configuration)
+    : ImageModel(model, configuration) {
     auto confidence_threshold_iter = configuration.find("confidence_threshold");
-    float confidence_threshold = 0.5;
     if (confidence_threshold_iter == configuration.end()) {
-        confidence_threshold = stof(model->get_rt_info<std::string>("model_info", "confidence_threshold"));
-    } else {
-        confidence_threshold = confidence_threshold_iter->second.as<float>();
-    }
-    auto labels_iter = configuration.find("labels");
-    std::vector<std::string> labels;
-    if (labels_iter == configuration.end()) {
-        labels = split(model->get_rt_info<std::string>("model_info", "labels"), ' ');
+        confidenceThreshold = stof(model->get_rt_info<std::string>("model_info", "confidence_threshold"));
     } else {
-        labels = labels_iter->second.as<std::vector<std::string>>();
-    }
-    auto layout_iter = configuration.find("layout");
-    std::string layout;
-    if (layout_iter != configuration.end()) {
-        layout = layout_iter->second.as<std::string>();
-    }
-    auto auto_resize_iter = configuration.find("auto_resize");
-    bool auto_resize = false;
-    if (auto_resize_iter != configuration.end()) {
-        auto_resize = auto_resize_iter->second.as<bool>();
-    }
-    auto iou_t_iter = configuration.find("iou_t");
-    float iou_t = 0.5;
-    if (iou_t_iter != configuration.end()) {
-        iou_t = iou_t_iter->second.as<bool>();
-    }
-    auto anchors_iter = configuration.find("anchors");
-    std::vector<float> anchors;
-    if (anchors_iter != configuration.end()) {
-        anchors = anchors_iter->second.as<std::vector<float>>();
-    }
-    auto masks_iter = configuration.find("masks");
-    std::vector<int64_t> masks;
-    if (masks_iter != configuration.end()) {
-        masks = masks_iter->second.as<std::vector<int64_t>>();
+        confidenceThreshold = confidence_threshold_iter->second.as<float>();
     }
-    auto resize_type_iter = configuration.find("resize_type");
-    std::string resize_type = "standard";
-    if (resize_type_iter == configuration.end()) {
-        if (model->has_rt_info("model_info", "resize_type")) {
-            resize_type = model->get_rt_info<std::string>("model_info", "resize_type");
-        }
-    } else {
-        resize_type = resize_type_iter->second.as<std::string>();
+}
+
+std::unique_ptr<DetectionModel> DetectionModel::create_model(const std::string& modelFile, std::string model_type, const ov::AnyMap& configuration) {
```

Comment:
Can we have another overload this function to take the inference adapter as a parameter?
`std::unique_ptr<DetectionModel> DetectionModel::create_model(InferenceAdapter adapter)`
in case of ovms it will be used like this 
`auto adapter=OVMS::getAdapter()`
`auto model=DetectionModel::create_model(adapter)`

The same for semgmentation_model and classification_model.

Reply:
Good idea. I hope that `OVMSInferenceAdapter` can return the model config from `.getModelConfig()` where we can grab the `model_type`. @Radwan-Ibrahim, what do you think?

Reply:
I updated the PR. Please take a look.
Added the same logic for Classification Model.

Reply:
Thanks Alexander, looks very good to me, I agree` OVMSInferenceAdapter ` should return the model config.